content/docs/command-reference/commit.md

```diff
@@ -67,17 +65,20 @@ Note that it's best to try avoiding these scenarios, where the
 <abbr>cache</abbr>, `dvc.lock`, and `.dvc` files are force-updated. DVC can't
 guarantee reproducibility in those cases.
 
+[add the missing deps/outs]: /docs/user-guide/how-to/add-deps-or-outs-to-a-stage
+
 ## Options
 
-- `-d`, `--with-deps` - determines files to commit by tracking dependencies to
-  the target stages or `.dvc` files. If no `targets` are provided, this option
-  is ignored. By traversing all stage dependencies, DVC searches backward from
-  the target stages in the corresponding pipelines. This means DVC will not
-  commit files referenced in later stages than the `targets`.
+- `-d`, `--with-deps` - only meaningful when specifying `targets`. This
+  determines files to commit by resolving all dependencies of the target stages
+  or `.dvc` files: DVC searches backward from the targets in the corresponding
+  pipelines. This will not commit files referenced in later stages than the
```

Comment:
Question: does this imply a kind of reclusiveness as well? In that stage dependencies are resolved recursively (not just once level before target stages). And if so should this be specified? I'd prefer to avoid using the term "recursive" here in such note though, not to create confusion with `-R`.

Reply:
What reclusiveness? :)

Reply:
Oops meant recursiveness. Stage dependencies are resolved recursively.